garage/envs/mujoco/mujoco_env.py

```diff
@@ -233,3 +235,10 @@ def print_stats(self):
 
     def action_from_key(self, key):
         raise NotImplementedError
+
+    @cached_property
+    def spec(self):
```

Comment:
`.spec` is not part of the [core `gym.Env`](https://github.com/openai/gym/blob/master/gym/core.py#L11) interface which is why it's not implemented here. That is, software assuming a `gym.Env` interface cannot necessarily assume that the object implementing that interface implements `spec`. This is well-documented in the comments on the `gym.Env` documentation.

If you have written software with which you hope to be able to use any environment implementing `gym.Env`, I recommend you rewrite it to not assume that `.spec` exists.

If you'd like to add this feature to garage, please add it throughout the whole codebase (i.e. all environments and wrappers). It is best suited as a separate PR from this one, since it's a new feature (not a bug fix).